### PR TITLE
Update spring boot

### LINKED ISSUE #6: update Spring Boot to latest stable release (2.0.1)
The project runs with Spring Boot 1.4.4. We should update it to the latest stable release.

### PR DESCRIPTION
**Pull request by lleifermann**
> Closes #6 

I updated spring boot to 2.0.1 and fixed an array of now broken things. This includes:
- findOne is no longer usable. Instead you now have to use findOneById which now returns an optional. Please especially look over those occurences where i tried to fix this inside streams. 
- WebMvcConfigurerAdapter is now deprecated an can be replaced by an implementation of WebMvcConfigurer.
- This still needs to be changed but i wanted feedback first. The annotation NotBlank is  
deprecated too. Any substitutes for this?
-Since the Hikari Datasource is now standard, do we need the current implementation of it?

EDIT: 

- As @DennisVonDerBey pointed out. Its possible to substitute Notblank from javax.validation.constraints. I did that accordingly. 
- The AbstractWebSocketMessageBrokerConfigurer was also deprecated and could be swapped with an implementation of WebSocketMessageBrokerConfigurer
- I removed all traces of our direct hikari implementation, since this is now directly supported by spring boot. I just deleted all our usages and it seems to work fine.

###### References

* [Spring-Boot 2.0 Migration Guide][sb2-mig-guide]
* [Spring Boot Security 2.0 Migration Guide][sb2-sec-mig-guide]

[sb2-mig-guide]: https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-2.0-Migration-Guide
[sb2-sec-mig-guide]: https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-Security-2.0

### CODE REVIEW

File: src/main/java/de/communicode/communikey/service/KeyService.java [hunk 1]
```diff
@@ -32,15 +32,12 @@
 import de.communicode.communikey.domain.User;
 import de.communicode.communikey.domain.UserGroup;
 import de.communicode.communikey.domain.UserEncryptedPassword;
-import de.communicode.communikey.exception.HashidNotValidException;
-import de.communicode.communikey.exception.KeyNotAccessibleByUserException;
-import de.communicode.communikey.exception.UserEncryptedPasswordNotFoundException;
+import de.communicode.communikey.exception.*;
```
Comment (fermentfan) [inline src/main/java/de/communicode/communikey/service/KeyService.java]: No asterisk import

File: src/main/java/de/communicode/communikey/service/UserService.java [hunk 2]
```diff
@@ -437,13 +439,16 @@ public User updateAuthorities(String login, Set<String> payload) {
         return ofNullable(userRepository.findOneByLogin(login))
             .map(user -> {
                 Set<Authority> payloadAuthorities = payload.stream()
-                    .map(authorityRepository::findOne)
+                    .map(id -> authorityRepository.findById(id).orElseThrow(() -> new AuthorityNotFoundException(id)))
```
Comment (fermentfan) [inline src/main/java/de/communicode/communikey/service/UserService.java:442]: Is this really necessary? Shouldn't it suffice to change the method?

File: src/main/java/de/communicode/communikey/service/UserService.java [hunk 3]
```diff
@@ -437,13 +439,16 @@ public User updateAuthorities(String login, Set<String> payload) {
...
                     .collect(Collectors.toSet());
                 if (!payloadAuthorities.equals(user.getAuthorities())) {
                     user.removeAllAuthorities();
                     payload.stream()
-                        .map(authorityRepository::findOne)
-                        .forEach(user::addAuthority);
+                        .map(authorityRepository::findById)
```
Comment (fermentfan) [inline src/main/java/de/communicode/communikey/service/UserService.java]: Duplicate code, already on it

File: src/main/java/de/communicode/communikey/security/SecurityAuditorAware.java [hunk 4]
```diff
@@ -43,11 +45,11 @@ public SecurityAuditorAware(CommunikeyProperties communikeyProperties) {
...
     @Override
-    public String getCurrentAuditor() {
+    public Optional<String> getCurrentAuditor() {
         String userName = SecurityUtils.getCurrentUserLogin();
         if (userName != null && !Objects.equals(userName, "anonymousUser")) {
-            return userName;
+            return ofNullable(userName);
         }
-        return communikeyProperties.getSecurity().getRoot().getLogin();
+        return ofNullable(communikeyProperties.getSecurity().getRoot().getLogin());
```
Comment (fermentfan) [inline src/main/java/de/communicode/communikey/security/SecurityAuditorAware.java]: ```return Optional.empty();```

**Comment by svengreb** [comment]
Changed the target branch to the new default branch `develop`.

File: src/main/java/de/communicode/communikey/config/DataSourceConfig.java [hunk 5]
```diff
@@ -17,91 +17,69 @@
...
  *
  * @author sgreb@communicode.de
  * @since 0.2.0
  */
 @Configuration
 @EnableJpaRepositories("de.communicode.communikey.repository")
-@EnableJpaAuditing(auditorAwareRef = "securityAuditorAware")
-@EnableTransactionManagement
+@EnableJpaAuditing(auditorAwareRef = "securityAuditorAware", dateTimeProviderRef="dateTimeProvider")
 public class DataSourceConfig {
```
Comment (svengreb) [inline src/main/java/de/communicode/communikey/config/DataSourceConfig.java]: Restored and updated in f5a271c6da073e532d72db82c85e2c6e9c8d1968.

File: src/main/java/de/communicode/communikey/exception/GlobalControllerExceptionHandler.java [hunk 6]
```diff
@@ -163,13 +163,11 @@
     public ErrorAttributes errorAttributes() {
```
Comment (svengreb) [inline src/main/java/de/communicode/communikey/exception/GlobalControllerExceptionHandler.java:163]: Updated in 7e6ad6831b5a177f0d1731bf0e97f6448e88bb57.

**Review by svengreb — APPROVED** [review]
A detailed documentation will be available in the next version release changelog. Also of course the changes are documented in the commit messages.
We are aware that the CI builds are failing which will be resolved in #2 and #8.

**Review by fermentfan — APPROVED** [review]

**Pull request merged**